nixos/modules/services/monitoring/datadog-agent.nix

```diff
@@ -201,6 +201,16 @@ in {
           excluded_interfaces = [ "lo" "lo0" ]; } ];
       };
     };
+
+    permissions = mkOption {
+      description = "User and group permissions for datadog-agent";
+      type = types.attrs;
```

Comment:
Please use a submodule so the type documents itself or switch to flat options instead of this nested approach.

Reply:
updated to use a submodule. let me know if there is something more you'd like in here.